Changed all British spellings in the Extending and Reference section to American spellings.

Buraah opened the pull request:
This pull request resolves [#10608](https://github.com/wagtail/wagtail/issues/10608)

I have changed all the British Spellings to American spellings in the Extending and Reference section of the documentation. I have also fixed minor grammatical and punctuation errors in these sections.

<!-- Thanks for contributing to Wagtail! 🎉  Please add a description below, explaining the purpose of this pull request - including the issue number of the issue you're fixing (if applicable). -->

Fixes #...







_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

lb- commented:
Hey @Buraah as per Slack. You'll probably need to rebase and resolve the conflicts. 

Also the CI has the following error. You may need to search through the docs for references to names or refs you've renamed. 

```
home/docs/checkouts/readthedocs.org/user_builds/wagtail/checkouts/11567/docs/releases/1.9.rst:58: WARNING: unknown document: '/reference/contrib/forms/customisation'
/home/docs/checkouts/readthedocs.org/user_builds/wagtail/checkouts/11567/docs/releases/2.14.rst:19: WARNING: undefined label: 'customising_group_views'
/home/docs/checkouts/readthedocs.org/user_builds/wagtail/checkouts/11567/docs/releases/5.2.md:309: WARNING: 'myst' cross-reference target not found: 'customising_group_views' [myst.xref_missing]
```

lb- commented on docs/advanced_topics/privacy.md:
```diff
@@ -4,9 +4,9 @@
--   **Accessible to any logged-in users:** The user must log in to view the page. All user accounts are granted access, regardless of permission level.
--   **Accessible with a shared password:** The user must enter the given shared password to view the page. This is appropriate for situations where you want to share a page with a trusted group of people, but giving them individual user accounts would be overkill. The same password is shared between all users, and this works independently of any user accounts that exist on the site.
--   **Accessible to users in specific groups:** The user must be logged in, and a member of one or more of the specified groups, in order to view the page.
+-   **Accessible to logged-in users:** The user must log in to view the page. All user accounts are granted access, regardless of permission level.
```
These names are intentionally worded to match the UI. Please leave as is. 

See https://github.com/wagtail/wagtail/pull/11546 for context.

lb- commented on docs/extending/adding_reports.md:
```diff
@@ -49,7 +49,7 @@ class UnpublishedChangesReportView(PageReportView):
 which provides an empty report page layout; for ``PageReportView``, this defaults to
 ``"wagtailadmin/reports/base_page_report.html"`` which provides a listing based on the explorer views,
-displaying action buttons, as well as the title, time of the last update, status, and specific type of any pages.
+displaying action buttons, as well as the title, time of the last update, status, and specific type of pages.
```
Not sure on this change. We are referring to a list of things and then an optional set of pages.

lb- commented on docs/extending/extending_client_side.md:
```diff
@@ -25,21 +25,21 @@ The simplest way is to add global JavaScript files via hooks, see [](insert_edit
 For JavaScript added when a specific Widget is used you can add an inner `Media` class to ensure that the file is loaded when the widget is used, see [Django's docs on their form `Media` class](https://docs.djangoproject.com/en/stable/topics/forms/media/#assets-as-a-static-definition).
 
-In a similar way, Wagtail's [](./template_components) provide a `media` property or `Media` class to add scripts when rendered.
+In a similar way, Wagtail's [](./template_components) provides a `media` property or `Media` class to add scripts when rendered.
```
I think the singular form is correct here. Assuming the title for the template component is 'template components'. 

This will currently render as 'Wagtail's template components provide a media property...'. 

I think this is better than  'Wagtail's template components provides a media property...'.

lb- commented on docs/extending/rich_text_internals.md:
```diff
@@ -203,11 +203,11 @@ The editor interface used on rich text fields can be configured with the [WAGTAI
 It is possible to create your own rich text editor implementation. At minimum, a rich text editor is a Django **_class_ django.forms.Widget** subclass whose constructor accepts an `options` keyword argument (a dictionary of editor-specific configuration options sourced from the `OPTIONS` field in `WAGTAILADMIN_RICH_TEXT_EDITORS`), and which consumes and produces string data in the HTML-like forma ...
 
-Typically, a rich text widget also receives a `features` list, passed from either `RichTextField` / `RichTextBlock` or the `features` option in `WAGTAILADMIN_RICH_TEXT_EDITORS`, which defines the features available in that instance of the editor (see [rich text features](rich_text_features)). To opt in to supporting features, set the attribute `accepts_features = True` on your widget class; the w ...
+Typically, a rich text widget also receives a `features` list, passed from either `RichTextField` / `RichTextBlock` or the `features` option in `WAGTAILADMIN_RICH_TEXT_EDITORS`, which defines the features available in that instance of the editor (see [rich text features](rich_text_features)). To opt into supporting features, set the attribute `accepts_features = True` on your widget class; the wi ...
```
I think the original is correct. 

https://english.stackexchange.com/questions/422848/opt-into-vs-opt-in-to
https://forum.wordreference.com/threads/opt-in-or-opt-into.3560327/

lb- replied:
I also could be wrong here, need some grammar gurus.

gasman replied:
I agree - "opt in" works as a phrasal verb, and the "in" part shouldn't be merged with the following word (just as we would tend to say "you must log in to the system" rather than "you must log into the system").

lb- commented on docs/reference/contrib/forms/customization.md:
```diff
@@ -764,11 +764,11 @@ class FormPage(AbstractEmailForm):
--   Each time a new `FormField` is added a `clean_name` also gets generated based on the user entered `label`.
--   `AbstractFormField` has a method `get_field_clean_name` to convert the label into a HTML valid `lower_snake_case` ASCII string using the [AnyAscii](https://pypi.org/project/anyascii/) library which can be overridden to generate a custom conversion.
+-   Each time a new `FormField` is added a `clean_name` also gets generated based on the user-entered `label`.
+-   `AbstractFormField` has a method `get_field_clean_name` to convert the label into an HTML valid `lower_snake_case` ASCII string using the [AnyAscii](https://pypi.org/project/anyascii/) library which can be overridden to generate a custom conversion.
```
I think `a` (the original) is more correct. 

https://www.scribbr.com/commonly-confused-words/a-vs-an/

gasman replied:
I disagree - in formal English the H in HTML is pronounced "aitch", not "haitch" (even if that's not true for all accents...) so the "an" is needed to match that. I'd also put a hyphen in "HTML-valid" though.

lb- commented on docs/reference/hooks.md:
```diff
@@ -458,13 +458,13 @@ def construct_forms_for_user(user, queryset):
 ### `register_rich_text_features`
 
-Rich text fields in Wagtail work with a list of 'feature' identifiers that determine which editing controls are available in the editor, and which elements are allowed in the output; for example, a rich text field defined as `RichTextField(features=['h2', 'h3', 'bold', 'italic', 'link'])` would allow headings, bold / italic formatting and links, but not (for example) bullet lists or images. The ` ...
+Rich text fields in Wagtail work with a list of 'feature' identifiers that determine which editing controls are available in the editor, and which elements are allowed in the output; for example, a rich text field defined as `RichTextField(features=['h2', 'h3', 'bold', 'italic', 'link'])` would allow headings, bold/italic formatting, and links, but not (for example) bullet lists or images. The `r ...
```
I prefer spaces between slashes, it wraps nicer, but not going to push either way unless there are others who think we should leave this.

lb- commented on docs/reference/signals.md:
```diff
@@ -186,11 +186,11 @@ This signal is emitted from a `TaskState` when a page's task is approved
 -   `kwargs` - Any other arguments passed to `task_approved.send()`
 
-## task_cancelled
+## task_canceled
```
Double check this is not an actual variable or method name.

lb- commented on docs/reference/signals.md:
```diff
@@ -150,13 +150,13 @@ This signal is emitted from a `WorkflowState` when a page's workflow completes s
 
-## workflow_cancelled
+## workflow_canceled
 
```
Double check this is not an actual variable or method name.

lb- commented on docs/reference/settings.md:
```diff
@@ -145,8 +145,8 @@ Note that all languages that exist in ``WAGTAIL_CONTENT_LANGUAGES`` must also ex
 
-Wagtail supports generating embed code from URLs to content on an external providers such as Youtube or Twitter. By default, Wagtail will fetch the embed code directly from the relevant provider's site using the oEmbed protocol.
-Wagtail has a builtin list of the most common providers.
+Wagtail supports generating embed code from URLs to content on external providers such as YouTube or X (Formally Twitter). By default, Wagtail will fetch the embed code directly from the relevant provider's site using the oEmbed protocol.
```
```suggestion
Wagtail supports generating embed code from URLs to content on external providers such as YouTube or X (formerly Twitter). By default, Wagtail will fetch the embed code directly from the relevant provider's site using the oEmbed protocol.
```

Lower case as it's not a proper noun. 

Should be formerly (past tense), not formally (as in to be more official).

lb- commented on docs/reference/pages/model_recipes.md:
````diff
@@ -276,4 +276,4 @@ class TagsSnippetViewSet(SnippetViewSet):
 ```
 
-A `Tag` model has a `name` and `slug` required fields. If you decide to add a tag, it is recommended to only display the `name` field panel as the slug field is autofilled when the `name` field is filled and you don't need to enter the same name in both the fields.
+A `Tag` model has a `name` and `slug` required fields. If you decide to add a tag, it is recommended to only display the `name` field panel as the slug field is auto-filled when the `name` field is filled and you don't need to enter the same name in both fields.
````
```suggestion
A `Tag` model has a `name` and `slug` required fields. If you decide to add a tag, it is recommended to only display the `name` field panel as the slug field is automatically populated when the `name` field is filled and you don't need to enter the same name in both fields.
```

Not sure on the word but we maybe should avoid the term autofill/auto-fill as it could be confused with browser auto-fill/autocomplete.

lb- reviewed (changes requested):
A few suggestions and items of feedback. Thank you for this.

Buraah commented:
> Hey @Buraah as per Slack. You'll probably need to rebase and resolve the conflicts.
> 
> Also the CI has the following error. You may need to search through the docs for references to names or refs you've renamed.
> 
> ```
> home/docs/checkouts/readthedocs.org/user_builds/wagtail/checkouts/11567/docs/releases/1.9.rst:58: WARNING: unknown document: '/reference/contrib/forms/customisation'
> /home/docs/checkouts/readthedocs.org/user_builds/wagtail/checkouts/11567/docs/releases/2.14.rst:19: WARNING: undefined label: 'customising_group_views'
> /home/docs/checkouts/readthedocs.org/user_builds/wagtail/checkouts/11567/docs/releases/5.2.md:309: WARNING: 'myst' cross-reference target not found: 'customising_group_views' [myst.xref_missing]
> ```

Okay, thank you @lb- 
I will start resolving this.
And yes, I renamed a file because it had the British spelling.
I am going to rebase, undo the file name changes, implement your feedback, and send another pull request.

lb- commented:
@Buraah You will be OK to rebase your current branch, you will need to `force push` to your remote branch when done.

If you are unsure on this, you can always create a backup of your branch locally (checkout a new branch with a similar name, then checkout the original branch again).

I find that a GUI tool like https://git-fork.com/ can really help make this easier. Remember you want to check out your branch and rebase 'on top of' the latest `main` from Wagtail.

If you are really stuck, I can rebase for you and force push to your remote branch.

Buraah commented:
> @Buraah You will be OK to rebase your current branch, you will need to `force push` to your remote branch when done.
> 
> If you are unsure on this, you can always create a backup of your branch locally (checkout a new branch with a similar name, then checkout the original branch again).
> 
> I find that a GUI tool like https://git-fork.com/ can really help make this easier. Remember you want to check out your branch and rebase 'on top of' the latest `main` from Wagtail.
> 
> If you are really stuck, I can rebase for you and force push to your remote branch.

Okay, thank you @lb-

Buraah commented:
Hello @lb- 
I have done the git rebase but I'm still getting the build error.
I tried to track the file with the error and I cannot seem to find what is wrong with it, as I didn't also make any new changes to the said file. 

This is the error: WARNING: 'myst' cross-reference target not found: '/advanced_topics/customization/page_editing_interface' [myst.xref_missing]

lb- commented on docs/topics/pages.md:
```diff
@@ -142,6 +142,7 @@ These are used for structuring fields in the interface.
 #### Customizing the page editor interface
 
+The page editor can be customized further. See [Customizing the editing interface](/advanced_topics/customization/page_editing_interface).
 The page editor can be customized further. See [Customizing the editing interface](/advanced_topics/customisation/page_editing_interface).
```
```suggestion
```

Looks like you added a new line here but did not remove the original.

This is probably why the CI is failing.

Buraah replied:
Thank You! @lb-, this was very helpful and all tests have finally passed.

lb- replied:
Awesome. Let us know when it's ready for another review.

Buraah commented:
> A few suggestions and items of feedback. Thank you for this.

I've implemented all feedback and it's now ready for review.

lb- reviewed (approved):
OK I think this is good to go.

I have rebased your branch locally and worked through the additional conflicts.

I might recommend you look into how to rebase, it's a tricky thing but really will help for these kinds of PRs that are working on top of a large surface area of changes.

You had 'merged' main into your branch and resolved things but this is not the same as rebasing.

I have also updated the last 3 pages with ~customise in their filename to be ~customize.

I will force push and check the CI and merge if all good.

Buraah commented:
> OK I think this is good to go.
> 
> I have rebased your branch locally and worked through the additional conflicts.
> 
> I might recommend you look into how to rebase, it's a tricky thing but really will help for these kinds of PRs that are working on top of a large surface area of changes.
> 
> You had 'merged' main into your branch and resolved things but this is not the same as rebasing.
> 
> I have also updated the last 3 pages with ~customise in their filename to be ~customize.
> 
> I will force push and check the CI and merge if all good.

Thank you for always taking the time to review and leave valuable feedback @lb-, It is well appreciated.
Since I'm still working on this issue, I will take the time to look into rebasing. I'm sure it will be relevant for other projects too.
Thank you again.
